src/test/java/seedu/sellsavvy/model/order/OrderListTest.java

```diff
@@ -45,6 +45,24 @@ public void contains_orderWithSameItemInList_returnsTrue() {
         assertTrue(orderList.contains(editedAtlas));
     }
 
+    @Test
+    public void hasPendingDuplicateOrderOf_nullOrder_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> orderList.hasPendingDuplicateOrderOf(null));
+    }
+
+    @Test
+    public void hasPendingDuplicateOrderOf_hasPendingDuplicate_returnsTrue() {
```

Comment:
this method name a bit long and confusing especially the part where hasPendingDuplicate appears twice.Took me quite a while to understand what u mean. Maybe merge all 3 test  for hasPendingDuplicate and split them with comments like what u do for isPendingDuplicateOf which u did below. 
![image](https://github.com/user-attachments/assets/fa474e8a-b3d1-45dd-bde4-56b28abb84f3)

Reply:
The other methods in OrderListTest have the same format of one test case per method especially since we need orderList to have no orders at the start of the test case. 
Also, I made a mistake for `hasPendingDuplicateOrderOf_noPendingDuplicate_returnFalse()` so I believe that made things more confusing than it's supposed to be.

Reply:
Implementation has been changed to use equals method so this method won't be used anymore.